### PR TITLE
[WIP] generate CJS bundles

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**

### CODE REVIEW

**Comment by pkozlowski-opensource** [comment]
This is WIP for CJS bundles generated with WebPack. There are several questions that need to be answered before we can push those bundles to users:
* do we bundle RxJS?
* what about external dependencies (Zone.js, polyfills for Reflect etc.)

This needs more work:
* depends on https://github.com/angular/angular/pull/4648
* needs router bundle
* testing, testing with different CJS-based workflows

**Comment by pkozlowski-opensource** [comment]
@vsavkin if you could give it a try and test with your WebPack workflow, this would be awesome!

File: package.json [hunk 1]
```diff
@@ -134,6 +134,7 @@
     "typescript": "^1.6.2",
     "vinyl": "^0.4.6",
     "walk-sync": "^0.1.3",
+    "webpack": "^1.12.2",
```
Comment (rkirov) [inline package.json]: don't forget the npm shrinkwrap dance. Follow https://github.com/angular/angular/blob/master/npm-shrinkwrap.readme.md.

File: gulpfile.js [hunk 2]
```diff
@@ -1179,6 +1179,39 @@ gulp.task('bundles.js', [
...
+        packageAlias: '' //this option is added to ignore "broken" package.json in our dist folder
+    },
+    output: {
+        filename: 'dist/js/bundle/angular2.cjs.dev.js',
+        libraryTarget: 'commonjs2'
+    }
+  }, done);
+});
+
+gulp.task('bundles.cjs', ['bundles.cjs.angular2', 'bundles.cjs.http']);
```
Comment (rkirov) [inline gulpfile.js]: Don't forget to hook into the other bundle building task - `bundle.js`.

**Comment by rkirov** [comment]
Looks good so far. You can try replacing the system builder bundle with the cjs bundle by editing https://github.com/angular/angular/blob/f7b75330e020d942be21fa704b02c57ec59a4f59/tools/broccoli/html-replace/SCRIPTS.html#L17 and rerunning the e2e tests.

**Comment by pkozlowski-opensource** [comment]
@rkirov thnx for your input. Will continue with this PR.

> You can try replacing the system builder bundle with the cjs bundle by editing https://github.com/angular/angular/blob/f7b75330e020d942be21fa704b02c57ec59a4f59/tools/broccoli/html-replace/SCRIPTS.html#L17 and rerunning the e2e tests.

Hmm, this might be more complex than this as we can't just load a CJS bundle via the `<script>` tag...

File: gulpfile.js (hunk 2 above)
Comment (pkozlowski-opensource) [inline gulpfile.js]: Yep!

File: package.json (hunk 1 above)
Comment (pkozlowski-opensource) [inline package.json]: Yeh, on my TODO list, thnx for the remainder!

**Comment by pkozlowski-opensource** [comment]
Updated the PR to produce bundles for the router as well and plug into the existing build. Unfortunately, the bundles that gets produced here are far from being ideal:
* rx gets duplicated in both `angular2` and `http` - we should decide if it should be an external dependency or not
* the `angular2` bundle doesn't contain mandatory deps (zone, Reflect polyfills) as those were pulling tons of other dependencies
*  we still duplicate facades in each and every bundle (additionally, router has duplicated DOM and some other decorator utils)

On top of this we start to have massive number of those bundles, since we do them for:
* 3 formats (CJS, System.register, SFX)
* dev and prod
* minified and not

This kind of explodes the number of combinations and will leave our users wondering what is going on....

**Comment by mhevery** [comment]
We are trying to keep our PR queue clean. This PR seems like WIP where no updates have happened for a while. Can we close it, and reopen once we are ready to continue?

**Comment by pkozlowski-opensource** [comment]
Ok, right, will close and re-open when ready for review. Working on this atm.

**Pull request closed without merging**